windows/src/engine/keyman32/CoreEnvironment.cpp

```diff
@@ -0,0 +1,32 @@
+#include "pch.h"
+
+BOOL SetupCoreEnvironment(km_kbp_option_item **core_environment) {
+  km_kbp_option_item *items = new km_kbp_option_item[5];
+
+  items[0].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[0].key = KM_KBP_KMX_ENV_BASELAYOUT;
+  items[0].value = reinterpret_cast<km_kbp_cp*>(Globals::get_BaseKeyboardName());
+
+  items[1].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[1].key = KM_KBP_KMX_ENV_BASELAYOUTALT;
+  items[1].value = reinterpret_cast<km_kbp_cp*>(Globals::get_BaseKeyboardNameAlt());
+
+  items[2].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[2].key = KM_KBP_KMX_ENV_SIMULATEALTGR;
+  items[2].value = Globals::get_SimulateAltGr() ? u"1" : u"0";
```

Comment:
Do we have to set `KM_KBP_KMX_ENV_SIMULATEALTGR` and `KM_KBP_KMX_ENV_BASELAYOUTGIVESCTRLRALTFORRALT` on Linux as well?

Maybe not since on Linux we add options from dconf, so if it's stored there we're fine.

Reply:
The `KM_KBP_KMX_ENV_SIMULATEALTGR` option is useful on Linux, but I am not sure if we surface it for the user?

`KM_KBP_KMX_ENV_BASELAYOUTGIVESCTRLRALTFORRALT` is probably only helpful on Windows where the Windows base keyboard may emit LCtrl+RAlt when RAlt is pressed, for some keyboards.

Reply:
>The KM_KBP_KMX_ENV_SIMULATEALTGR option is useful on Linux, but I am not sure if we surface it for the user?

Yes, that doesn't currently surface for the user - unless he does the equivalent of a registry hack and sets the dconf setting manually...

Reply:
Future feature for Keyman for Linux: make the Simulate AltGr option available (and then we need to test it thoroughly I guess!)